src/ArmTemplates/Extractor/EntityExtractors/ProductExtractor.cs

```diff
@@ -157,5 +147,33 @@ public class ProductExtractor : IProductExtractor
                 throw;
             }
         }
+
+        async Task AddGroupsLinkedToProductResources(
```

Comment:
IMO, maybe, it is not a part of this PR, but we can try to split all linked-entities logic in separate extractors. Because there is **api**-**operation** entities link, and we have a separate `ApiOperationExtractor` for that. I would prefer make this as maintainable as possible, so I would add `ProductGroupsExtractor`, which will only work with products + groups linked entity. 

Also we can take advantage of already loaded resources from client (products and groups). We can try to cache that resources in processor, i.e. But it is just a first approach, so I am not sure about that :)